vin/test/vin-client.rb

```diff
@@ -1,267 +0,0 @@
-    post_request url
-  end
-
-  def get_shipment_note uid, sid, nid
-    url = URI::join(@url,"subs/#{uid}/shipments/#{sid}/notes/#{nid}")
-    get_request url
-  end
-
-  def put_shipment_note uid, sid, nid, options = {}
-    url = URI::join(@url,"subs/#{uid}/shipments/#{sid}/notes/#{nid}")
-    read = options[:read]
-    if read.nil?
-      read_note
-    else
-      read_note read
-    end
-    put_request url
-  end
-
-  def delete_shipment_note uid, sid, nid
-    url = URI::join(@url,"subs/#{uid}/shipments/#{sid}/notes/#{nid}")
-    delete_request url
-  end
-
-  def get_wines uid
-    url = URI::join(@url,"subs/#{uid}/wines")
-    get_request url
-  end
-
-  def get_wine uid, wid
-    url = URI::join(@url,"subs/#{uid}/wines/#{wid}")
-    get_request url
-  end
-
-  def get_wine_notes uid, wid
-    url = URI::join(@url,"subs/#{uid}/wines/#{wid}/notes")
-    get_request url
-  end
-
-  def post_wine_notes uid, wid, options = {}
-    url = URI::join(@url,"subs/#{uid}/wines/#{wid}/notes")
-    read = options[:read]
-    if read.nil?
-      read_note
-    else
-      read_note read
-    end
-    post_request url
-  end
-
-  def get_wine_note uid, wid, nid
-    url = URI::join(@url,"subs/#{uid}/wines/#{wid}/notes/#{nid}")
-    get_request url
-  end
-
-  def put_wine_note uid, wid, nid, options = {}
-    url = URI::join(@url,"subs/#{uid}/wines/#{wid}/notes/#{nid}")
-    read = options[:read]
-    if read.nil?
-      read_note
-    else
-      read_note read
-    end
-    put_request url
-  end
-
-  def delete_wine_note uid, wid, nid
-    url = URI::join(@url,"subs/#{uid}/wines/#{wid}/notes/#{nid}")
-    delete_request url
-  end
-
-  def get_wine_rating uid, wid
-    url = URI::join(@url,"subs/#{uid}/wines/#{wid}/rating")
-    get_request url
-  end
-
-  def post_wine_rating uid, wid, options = {}
-    url = URI::join(@url,"subs/#{uid}/wines/#{wid}/rating")
-    read = options[:read]
-    if read.nil?
-      read_rating
-    else
-      read_rating read
-    end
-    post_request url
-  end
-
-  def get_user_delivery uid
-    url = URI::join(@url,"subs/#{uid}/delivery")
-    get_request url
-  end
-
-  def put_user_delivery uid, options = {}
-    url = URI::join(@url,"subs/#{uid}/delivery")
-    read = options[:read]
-    if read.nil?
-      # read some delivery example
-    else
-      # read some delivery example read
-    end
-    put_request url
-  end
-
-  def post_admin options = {}
-    url = URI::join(@url,"admins")
-    read = options[:read]
-    if read.nil?
-      read_admin
-    else
-      read_admin read
-    end
-    post_request url
-  end
-
-  def get_admins
-    url = URI::join(@url,"admins")
-    get_request url
-  end
-
-  def get_admin aid
-    url = URI::join(@url,"admins/#{aid}")
-    get_request url
-  end
-
-  def put_admin aid, options = {}
-    url = URI::join(@url,"admins/#{aid}")
-    read = options[:read]
-    if read.nil?
-      read_admin
-    else
-      read_admin read
-    end
-    put_request url
-  end
-
```

Comment:
Explain yourself. I actually wrote this in a hurry (using the sublime feature of multiple cursors in sublime text) and don't really know how to do it better... It's even worst because i took the original client from the first version of the project (Sinatra instead of Rails). But it should be fine since it is just a RestClient.
I want all my PUT/POST methods the hability to run using the default JSON readed from `lib/data`, but i also want them to read it from the user if another file is gieven (to test different imputs). For every entity (subscriber, wine, note, rating...) there are different files, but the methods just do the same, but with a different JSON object... Do you know a better way to ignore `nil` arguments in Ruby?